packages/checkout/widgets-lib/src/widgets/bridge/components/BridgeForm.tsx

```diff
@@ -227,24 +226,44 @@ export function BridgeForm(props: BridgeFormProps) {
     // fetching or the user is updating the inputs.
     if ((silently && (loading || editing)) || !checkout) return;
 
-    const bridgeFeeAction = from?.network === getL1ChainId(checkout.config)
-      ? BridgeFeeActions.DEPOSIT
-      : BridgeFeeActions.WITHDRAW;
-
-    const gasEstimate = await tokenBridge!.getFee({
-      action: bridgeFeeAction,
-      gasMultiplier: 1.1,
-      sourceChainId: from?.network.toString() ?? '',
-      destinationChainId: to?.network.toString() ?? '',
-      token: NATIVE.toUpperCase(),
-      amount: BigNumber.from(0),
-    });
+    const tokenIsNative = isNativeToken(formToken?.token.address);
+    const amountToBridge = tokenIsNative
+      ? utils.parseUnits(formAmount)
+      : utils.parseUnits(formAmount, formToken?.token.decimals);
+
+    const tokenAddress = tokenIsNative ? NATIVE.toUpperCase() : formToken?.token.address;
+
+    const [unsignedApproveTransaction, unsignedTransaction] = await Promise.all([
+      tokenBridge!.getUnsignedApproveBridgeTx({
+        senderAddress: from!.walletAddress!,
+        token: tokenAddress ?? NATIVE.toUpperCase(),
+        amount: amountToBridge,
+        sourceChainId: from!.network.toString(),
+        destinationChainId: to!.network.toString(),
+      }),
```

Comment:
Should we omit this if the transfer is for a native asset? (i.e. `transferIsNative` is `true`?)

Reply:
I guess we could, however, could we perhaps treat it as a separate PR because this will need further testing given that  I will need to make these changes across two screens. Right now I have copied the logic from a view that has already been tested.